ntbs-service/DataAccess/AlertRepository.cs

```diff
@@ -13,6 +13,7 @@ public interface IAlertRepository
         Task<Alert> GetAlertByIdAsync(int? alertId);
         Task<Alert> GetAlertByNotificationIdAndTypeAsync(int notificationId, AlertType alertType);
         Task<Alert> GetOpenAlertByNotificationIdAndTypeAsync(int notificationId, AlertType alertType);
+        Task<Alert> GetOpenAlertByNotificationId<T>(int notificationId) where T : Alert;

```

Comment:
yeee, i kinda through this in here as a "let's see what people think of using this" ... I've now merged the two, and converted the rest of the methods to care about the alert type through the type system, rather than explicitly :D